.github/workflows/snyk_scan.yaml

```diff
@@ -0,0 +1,26 @@
+---
+name: vanagon check PRs
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+ snyk_vanagon:
+   runs-on: ubuntu-latest
+   steps:
+    - name: checkout the current PR
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        persist-credentials: false
+    - name: Run Vanagon Snyk Scan
+      uses: puppetlabs/security-snyk-vanagon-action@v2
+      with:
+        snykToken: ${{ secrets.SNYK_PE_TOKEN }}
+        snykOrg: 'puppet-enterprise'
+        skipPlatforms: 'debian-10-armhf'
```

Comment:
you're missing some (probably) required args:

https://github.com/puppetlabs/security-snyk-vanagon-action/blob/main/sample_workflow.yaml#L27-L28

Reply:
@LivingInSyn is there a good iterative process for these (eg edit the file, trigger a manual action, make more edits, etc) but without merging? It's not clear to me if adding a `pull_request` event will run the workflow in *this* PR or the base branch.

Reply:
yes, you can add a  `workflow_dispatch` trigger instead of on `pull_request` while testing https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

Reply:
Hmm, that doesn't seem to work. The [manually running a workflow docs](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) says:

> To trigger the workflow_dispatch event, your workflow must be in the default branch.

To test this out, I pushed a test PR with `on: workflow_dispatch`, but the snyk workflow is not listed as a check.

Reply:
oh, right, sorry!

Reply:
@LivingInSyn thanks for the PR review. 

I added those missing args and switched to trigger on `workflow_dispatch`. Now that this is set to `workflow_dispatch`, it sounds like we can merge this PR and iterate from there?

Reply:
Based on https://stackoverflow.com/questions/67707244/how-to-test-github-workflow-without-merging-into-master-main-branch it looks like you can set `on: pull_request` and the PR will run the workflow from the PR even if it's not in the base branch. The author that replied has an example PR https://github.com/chenrui333/github-action-test/pull/33/files and the workflow checks ran in https://github.com/chenrui333/github-action-test/actions/runs/873013117

Reply:
Thanks @joshcooper . I've set the workflow to trigger `on: pull_request` and it looks like it's working in this PR.

However, it now seems to be having some kind of auth failure when the workflow runs. I'm seeing this error message: `couldn't setup the env varsno snyk token set`.

The `snykToken` is set in the workflow the same it is in other repos, and I see the `SNYK_PE_TOKEN` present in this repo's settings. Am I missing an additional auth step, or is there some issue in the workflow YAML?

Reply:
That's a good question... Perhaps the token isn't available to pull requests, otherwise a malicious user could submit a PR and steal the token.. But the PR would have to be approved first... Thoughts @LivingInSyn?

Reply:
If you are making a PR from a remote repo and using `pull_request` (not `pull_request_target`) then the secrets will not be available to the github action.

Reply:
Hmm, I changed the action to `pull_request_target` and I can't seem to get the workflow to trigger, even after closing/reopening the PR. 

I'm thinking about changing the action back to `workflow_dispatch`, merging, manually testing, then permanently setting the action to trigger on push to main (as that seems to be how most of our repos are configured).

Does that sound alright @LivingInSyn ? I'm otherwise having a hard time squaring the circle on getting this action to trigger as a PR while also being able to access the token.

Reply:
I'm thinking we'd need to push a branch directly to PL/upstream and make a PR from there. I'm also fine triggering on pull and push and merging it.

Reply:
Yes, it looks like a direct branch can trigger the workflow. I've opened https://github.com/puppetlabs/pxp-agent-vanagon/pull/43 and will continue over there. 

Closing out this PR